changelog/fragments/helm-conditional.yaml

```diff
@@ -0,0 +1,27 @@
+# entries is a list of entries to include in
+# release notes and/or the migration guide
+entries:
+  - description: >
+      The status conditional `UpdateError` was renamed to `UpgradeError` to better align with Helm nomenclature.
+      This change just affects Helm based-operators.
+
+    # kind is one of:
+    # - addition
+    # - change
+    # - deprecation
+    # - removal
+    # - bugfix
+    kind: "change"
+
+    # Is this a breaking change?
+    breaking: yes
+    
+    # Migration can be defined to automatically add a section to
+    # the migration guide. This is required for breaking changes.
+    migration:
+      header: `UpdateError` status for Helm based-Operator was ranamed 
+      body: >
+        For Helm based-Operator be more aligned with Helm the `UpdateError` was renamed to `UpgradeError`. 
+        Note that it is NOT a breaking change for the Helm based-Operators itself, however, if you have any 
+        script or code implemented which is using this status to perform checks then, will be required replace 
+        `UpdateError` for `UpgradeError` in this places. 
```

Comment:
Hi @estroz, 

What we are changing here is the **status name** which will appear in the CR.status  to be aligned with helm ( part of the "new" helm-operator ). I agree that users might be checking it due to a script, for example, the status for tests and etc and then, it would be a breaking change for them but not for o Helm based-Operator itself.

My idea/suggestion is we do it before release 1.0.0 image for users is able to move smoothly. 
@joelanford wdyt?